test/liquid-vault.js

```diff
@@ -378,16 +378,88 @@ describe('LiquidVault', function () {
+    const transferToLiquidVault = utils.parseUnits('20000', baseUnit); // 20.000 tokens
+
+    await infinity.transfer(liquidVault.address, transferToLiquidVault);
+    assertBNequal(await infinity.balanceOf(liquidVault.address), transferToLiquidVault);
+    
+    await liquidVault.purchaseLP({ value: purchaseValue });
+    const lockedLP = await liquidVault.getLockedLP(owner.address, 0);
+    const { donationShare } = await liquidVault.config();
+    const stakeDuration = await liquidVault.getStakeDuration();
+    const lpBalanceBeforeClaim = await uniswapPair.balanceOf(owner.address);
+
+    await ganache.setTime((bn(lockedLP[2]).add(stakeDuration)).toNumber());
+    const claimLP = await liquidVault.claimLP();
+    const receipt = await claimLP.wait();
+
+    const { holder, amount, exitFee, claimed } = receipt.events[0].args;
+    const estimatedFeeAmount = lockedLP[1].mul(donationShare).div(bn('100'));
+    const lpBalanceAfterClaim = await uniswapPair.balanceOf(owner.address);
+
+    assert.strictEqual(holder, owner.address);
+    assert.isTrue(claimed);
+    assertBNequal(amount, lockedLP[1]);
+    assertBNequal(exitFee, estimatedFeeAmount);
+    assertBNequal(amount.sub(exitFee), lpBalanceAfterClaim.sub(lpBalanceBeforeClaim));
+
+    uniswapPairUserBalanceBefore = await uniswapPair.balanceOf(user.address);
+    uniswapPairOwnerBalanceBefore = await uniswapPair.balanceOf(owner.address);
+
+    await uniswapPair.transfer(user.address, lpBalanceAfterClaim);
+
+    uniswapPairUserBalanceAfter = await uniswapPair.balanceOf(user.address);
+    uniswapPairOwnerBalanceAfter = await uniswapPair.balanceOf(owner.address);
+
+    assertBNequal(uniswapPairUserBalanceBefore, 0);
+    assertBNequal(uniswapPairOwnerBalanceAfter, 0);
+
+    await uniswapPair.approve(uniswapRouter.address, lpBalanceAfterClaim);
+    await expect(uniswapRouter.removeLiquidity(
+      infinity.address,
+      weth.address,
+      lpBalanceAfterClaim,
+      0,
+      0,
+      owner.address,
+      new Date().getTime() + 3000,
+    )).to.be.revertedWith('revert ds-math-sub-underflow');
+
+    const infinityBalanceBefore = await infinity.balanceOf(user.address);
+    const ethUserBalanceBefore = await ethers.provider.getBalance(user.address);
+
+    await uniswapPair.connect(user).approve(uniswapRouter.address, lpBalanceAfterClaim);
+    await expect(uniswapRouter.connect(user).removeLiquidityETH(
+      infinity.address, 
+      lpBalanceAfterClaim, 
+      0, 
+      0, 
+      user.address, 
+      new Date().getTime() + 3000,
+    )).to.emit(uniswapPair, 'Burn');
+
+    const infinityBalanceAfter = await infinity.balanceOf(owner.address);
+    const ethUserBalanceAfter = await ethers.provider.getBalance(user.address);
+    
+    assert.isTrue(infinityBalanceAfter.gt(infinityBalanceBefore));
+    assert.isTrue(ethUserBalanceAfter.gt(ethUserBalanceBefore));
+
+    const lpBalanceAfterRemoveLiquidity = await uniswapPair.balanceOf(owner.address);
```

Comment:
lpBalanceAfterRemoveLiquidity is zero from line 422. so You need to check that USER and not owner has LP tokens before removeLiquidityETH and doesn't have after transaction

Reply:
resolved